### PR TITLE
Update to spring boot 1.4

### PR DESCRIPTION
**Pull request by bjoernhaeuser**
@rebuy-de/prp-rebuy-silo-archetype

### CODE REVIEW

File: src/main/resources/archetype-resources/silo/pom.xml [hunk 1]
```diff
@@ -17,31 +17,29 @@
...
         <h2.version>1.4.190</h2.version>
         <hamcrest-all.version>1.3</hamcrest-all.version>
-        <jackson-databind.version>2.6.4</jackson-databind.version>
-        <javax.servlet-api.version>3.1.0</javax.servlet-api.version>
-        <json-path.version>2.1.0</json-path.version>
+        <jackson-databind.version>2.8.2</jackson-databind.version>
+        <json-path.version>2.2.0</json-path.version>
         <junit.version>4.12</junit.version>
-        <maven-apt-plugin.version>1.0.4</maven-apt-plugin.version>
+        <apt-maven-plugin.version>1.1.3</apt-maven-plugin.version>
```
Comment (das-vinculum) [inline src/main/resources/archetype-resources/silo/pom.xml]: Are we aware of the fact, that we cannot use Classes with it, that have AspectJ on board? 

See https://github.com/querydsl/apt-maven-plugin for the information.

**Comment by hendrik-rebuy** [comment]
we should think about having an automated test for this.

File: src/main/resources/archetype-resources/silo/pom.xml (hunk 1 above)
Comment (bjoernhaeuser) [inline src/main/resources/archetype-resources/silo/pom.xml]: We are already using that plugin, i just updated the version. So: yes.

**Comment by bjoernhaeuser** [comment]
@hendrik-rebuy tried to do that, the problem is that the generated silo does not start properly, because the database config is not setup to 100% (you still have to choose the driver). Easiest solution would be to use h2 by default.

File: src/main/resources/archetype-resources/silo/src/main/java/silo/configuration/LogConfiguration.java [hunk 2]
```diff
@@ -29,22 +26,17 @@
     private void configureLog4j()
```
Comment (AnkushKhanna) [inline src/main/resources/archetype-resources/silo/src/main/java/silo/configuration/LogConfiguration.java]: This method needs to be called in a Post Construct.

Comment (bjoernhaeuser) [inline src/main/resources/archetype-resources/silo/src/main/java/silo/configuration/LogConfiguration.java]: good catch, thanks

File: src/main/resources/archetype-resources/silo/src/main/resources/production/log4j2.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        </RollingFile>
+        <Console name="STDOUT" target="SYSTEM_OUT">
+            <PatternLayout pattern="%d %-5p [%t] %C{2} (%F:%L) - %m%n"/>
+        </Console>
+    </Appenders>
+    <Appenders>
+        <GELF name="gelfAppender" server="logstash-gelf.service.dc1.consul" port="9000" includeExceptionCause="true">
+            <PatternLayout pattern="%logger{36} - %msg%n"/>
+            <Filters>
+                <Filter type="MarkerFilter" marker="FLOW" onMatch="DENY" onMismatch="NEUTRAL"/>
```
Comment (AnkushKhanna) [inline src/main/resources/archetype-resources/silo/src/main/resources/production/log4j2.xml]: For me this did not work, had to use 
```
<Filters>
     <MarkerFilter marker="FLOW" onMatch="DENY" onMismatch="NEUTRAL"/>
     <MarkerFilter marker="EXCEPTION" onMatch="DENY" onMismatch="ACCEPT"/>
</Filters>
```

**Pull request merged**